docs/handcraft.rst

```diff
@@ -0,0 +1,2049 @@
+
+    In [0]: from sha3 import sha3_256
+
+    In [0]: json_str_tx = json.dumps(
+       ...:     handcrafted_creation_tx,
+       ...:     sort_keys=True,
+       ...:     separators=(',', ':'),
+       ...:     ensure_ascii=False,
+       ...: )
+
+    In [0]: txid = sha3_256(json_str_tx.encode()).hexdigest()
+
+    In [0]: handcrafted_creation_tx['id'] = txid
+    
+Compare to the txid generated via the function prepare_transaction:
+
+.. ipython::
+
+    In [0]: txid == prepared_creation_tx['id']
+
+You may observe that
+
+.. ipython::
+
+    In [0]: handcrafted_creation_tx == prepared_creation_tx
+
+.. ipython::
+
+    In [0]: from copy import deepcopy
+
+    In [0]: # back up
+
+    In [0]: prepared_creation_tx_bk = deepcopy(prepared_creation_tx)
+
+    In [0]: # set fulfillment to None
+
+    In [0]: prepared_creation_tx['transaction']['fulfillments'][0]['fulfillment'] = None
+ 
+    In [0]: handcrafted_creation_tx == prepared_creation_tx
+
+Still not equal because we used tuples instead of strings
+
+.. ipython::
+
+    In [0]: # serialize to json str
+
+    In [0]: json_str_handcrafted_tx = json.dumps(handcrafted_creation_tx, sort_keys=True)
+
+    In [0]: json_str_prepared_tx = json.dumps(prepared_creation_tx, sort_keys=True)
+
+.. ipython::
+
+    In [0]: json_str_handcrafted_tx == json_str_prepared_tx
+
+    In [0]: prepared_creation_tx = prepared_creation_tx_bk
+
+The full handcrafted yet to be fulfilled transaction payload:
+
+.. ipython::
+ 
+    In [0]: handcrafted_creation_tx
+
+
+The Fulfilled Transaction
+=========================
+
+.. ipython::
+ 
+    In [0]: from cryptoconditions.crypto import Ed25519SigningKey
+
+    In [0]: from bigchaindb_driver.offchain import fulfill_transaction
+
+    In [0]: fulfilled_creation_tx = fulfill_transaction(
+       ...:     prepared_creation_tx,
+       ...:     private_keys=alice.signing_key,
+       ...: )
+
+    In [0]: sk = Ed25519SigningKey(alice.signing_key)
+
+    In [0]: message = json.dumps(
+       ...:     handcrafted_creation_tx,
+       ...:     sort_keys=True,
+       ...:     separators=(',', ':'),
+       ...:     ensure_ascii=False,
+       ...: )
+
+    In [0]: ed25519.sign(message.encode(), sk)
+
+    In [0]: fulfillment = ed25519.serialize_uri()
+
+    In [0]: handcrafted_creation_tx['transaction']['fulfillments'][0]['fulfillment'] = fulfillment
+
+checks
+
+.. ipython::
+
+    In [0]: fulfilled_creation_tx['transaction']['fulfillments'][0]['fulfillment'] == fulfillment
+
+    In [0]: json.dumps(fulfilled_creation_tx, sort_keys=True) == json.dumps(handcrafted_creation_tx, sort_keys=True)
+
+
+In a nutshell
+=============
+
+Handcrafting a ``'CREATE'`` transaction can be done as follows:
+
+.. code-block:: python
+
+    import json
+    from uuid import uuid4
+
+    import sha3
+    import cryptoconditions
+    
+    from bigchaindb_driver.crypto import generate_keypair
+
+
+    alice = generate_keypair()
+
+    operation = 'CREATE'
+    
+    asset_id = str(uuid4())
```

Comment:
Well, yes, here because I did not care about comparing with the payloads generated by `prepare_transaction` and `fulfill_transaction`